rialto_airflow/funders/linker.py

```diff
@@ -90,23 +91,20 @@ def link_openalex_publications(snapshot) -> int:
 
         with get_session(snapshot.database_name).begin() as update_session:
             for funder in funders:
-                if openalex_funder := _lookup_openalex_funder(funder):
-                    logging.info(
-                        f"found funder data in openalex for {openalex_funder['id']}"
+                openalex_funder_id = funder["funder"]
+                if funder_id := _find_or_create_openalex_funder(
+                    update_session, openalex_funder_id
+                ):
+                    update_session.execute(
+                        insert(pub_funder_association)
+                        .values(publication_id=pub.id, funder_id=funder_id)
+                        .on_conflict_do_nothing()
                     )
-                    if funder_id := _find_or_create_funder(snapshot, openalex_funder):
-                        update_session.execute(
-                            insert(pub_funder_association)
-                            .values(publication_id=pub.id, funder_id=funder_id)
-                            .on_conflict_do_nothing()
-                        )
     logging.info(f"processed {count} publications from OpenAlex")
     return count
 
 
```

Comment:
`_find_or_create_funder` was split into `_find_or_create_dim_funder` and `_find_or_create_openalex_funder` since they need to do different things.